docs/build_from_source.md

```diff
@@ -0,0 +1,100 @@
+# Build BladeDISC from Source in a Docker container
+
+This document introduced how to build BladeDISC from source,
+to easy the software installation and configuration, we provide a
+[Dockerfile](/docker/dev/Dockerfile) that contains requirements software
+to build and test.
+
+## Prerequisite
+
+1. Git for checking out the source code.
+1. [Nvidia Docker](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html)
+to launch Docker container on Nvidia GPU host.
+
+## Checkout the Source
+
+``` bash
+git clone git@github.com:alibaba/BladeDISC.git
+cd BladeDISC && git submodule update --init --recursive
+```
+
+## Building BladeDISC for TensorFlow Wrapper
+
+1. step0: launch a development Docker container that runs a
+development Docker image.
+
+    ``` bash
+    nvidia-docker run --rm -it -v $PWD:/disc bladedisc/bladedisc:latest-devel-cuda11.0 bash
+    ```
+
+    please goto [this website](https://hub.docker.com/r/bladedisc/bladedisc/tags?page=1&name=devel) to
+    find more images with various CUDA versions.
+
+1. step1: configuration stage.
+
+    ``` bash
+    python scripts/python/tao_build.py /opt/venv_disc -s configure --bridge-gcc default --compiler-gcc default
+    ```
+
+    - `/opt/venv_disc` is a pre-installed Python virtual environment with requirement Python packages.
+    - `-s configure` specified the building stage.
+    - `--bridge-gcc` and `--compiler-gcc` specified the GCC version building with tao bridge and disc compiler core.
+
+1. step2: build tao bridge library
+
```

Comment:
New users may not know what 'tao' means. We need to modify the terms in `tao_build.py`.

Reply:
As the office discussion, using an bash script to build tensorflow_bladedisc.